### PR TITLE
Revise `SelectPanel` documentation for new version

### PR DESCRIPTION
**Pull request by dipree**
Preparing the select panel documentation for the new implementation.

- Streamline content to be less elaborate and easier to consume.
- Update the structure for best readability. Guidelines [here](https://primer.style/guides/contribute/documentation#component-documentation-structure) and [here](https://github.com/primer/design/blob/main/CONTRIBUTING.md#design-guidelines-template) are currently a little inconsistent. Going with what's adopted in most components:
  - Overview
  - Anatomy
  - Options
  - Behaviors (also called [Interactions](https://primer.style/guides/contribute/documentation#interactions))
  - Usage guidelines (we don't have that in the guidelines)
  - Accessibility
- Remove redundant content unless it helps to explain accessibility best practices.

Latest [deployment](https://primer-3de09d22a0-26441320.drafts.github.io/components/selectpanel)

### CODE REVIEW

File: content/components/selectpanel.mdx
Comment (maximedegreve) [inline content/components/selectpanel.mdx:24]: Kinda wish we still had this information. A lot of people do get confused about this.

Comment (maximedegreve) [reply content/components/selectpanel.mdx:24]: `the capability to make multiple selections` is kinda wrong here as menu's support this too.

File: content/components/selectpanel.mdx
Comment (siddharthkp) [inline content/components/selectpanel.mdx:17]: I like that you're mentioning these changes that people come looking for!

File: content/components/selectpanel.mdx
Comment (siddharthkp) [inline content/components/selectpanel.mdx:162]: Clarifying question: Are these intended as examples or a closed set?

File: content/components/selectpanel.mdx
Comment (siddharthkp) [inline content/components/selectpanel.mdx:193]: 👀

**Review by siddharthkp — APPROVED** [review]
Left some clarifying questions, approving in advance though

File: content/components/selectpanel.mdx
Comment (dipree) [reply content/components/selectpanel.mdx:24]: Incorporated it in the first paragraph now.

File: content/components/selectpanel.mdx
Comment (dipree) [reply content/components/selectpanel.mdx:162]: @siddharthkp good question, I didn't change anything to the existing content here, just condensed it a little. I assume that it was meant to be exclusive to these. Is there anything else that is missing here?

File: content/components/selectpanel.mdx
Comment (keithamus) [inline content/components/selectpanel.mdx:14]: I don't think this conveys the original prose. ActionMenu is different from SelectPanel, the two should not be conflated as they offer different use cases. I don't think it is sufficient to say SelectPanel offers more controls because it implies that ActionMenu could be used in its place, which is, IMO, not true. For example ActionMenu is immediately reactive (and so useful for non-destructive writes, e.g. filtering) while SelectPanel requires the user to make choices then commit.

File: content/components/selectpanel.mdx
Comment (keithamus) [inline content/components/selectpanel.mdx:50]: I'd like to encourage people towards actionmenu for _these_ instances. Unless the list is large enough to warrant a type-ahead filter, then we should use actionmenu. One that springs to mind is License chooser which has maybe 10 licenses, and can display the list all at once without requiring typeahead.

Comment (dipree) [reply content/components/selectpanel.mdx:50]: That makes sense. Since this section is focusing on the options of the component, we could create a new section under "[Usage guidelines](https://primer-3de09d22a0-26441320.drafts.github.io/components/selectpanel#usage-guidelines)" to point that out, thoughts?

File: content/components/selectpanel.mdx
Comment (dipree) [reply content/components/selectpanel.mdx:14]: > It is often used to augment information of a wide variety of items.

The initial phrasing is vague and the idea is to be more specific, but if it's not correct, that obviously doesn't help. Is there a good, distinct way of explaining the difference between the two to avoid any ambiguity here that you could think of?

**Comment by ichelsea** [comment]
I added myself as a reviewer here; looks like a quite a bit of changes from what I originally wrote and then worked with @maximedegreve on revising. 

I'd like to get this reviewed today and hoping we can hold off merging until next week on this one.

File: content/components/selectpanel.mdx
Comment (keithamus) [reply content/components/selectpanel.mdx:42]: Sounds like a good idea.

Comment (dipree) [reply content/components/selectpanel.mdx:42]: More prominent and probably the best place in the [doc](https://primer-3de09d22a0-26441320.drafts.github.io/components/selectpanel):

Screenshot showing the single select section with a info box that clarifies about action menu

File: content/components/selectpanel.mdx
Comment (dipree) [reply content/components/selectpanel.mdx:272]: Agreed it's redundant.

File: content/components/selectpanel.mdx
Comment (dipree) [reply content/components/selectpanel.mdx]: I see, brought back the initial paragraph with slight changes in wording.

File: content/components/selectpanel.mdx
Comment (dipree) [reply content/components/selectpanel.mdx]: Brought back the more detailed description but changed the order of it so that search comes first and "Select all" last. That way it's sorted by priority respectively likelihood of adoption scenario. Also swapped the examples in the image above that you provided me 🙇 to accompany that.

https://primer-3de09d22a0-26441320.drafts.github.io/components/selectpanel#list-multi-select

Screenshot of the rendered section with the changes described above

File: content/components/selectpanel.mdx
Comment (siddharthkp) [inline content/components/selectpanel.mdx:299]: (Not related to just this PR) Why is this the case @maximedegreve?

Also, this is only true for single selection, right? not multiple selection (because radio)

File: content/components/selectpanel.mdx
Comment (siddharthkp) [inline content/components/selectpanel.mdx:194]: Instead of removing this section completely, should we replace this with the behavior we want now?

File: content/components/selectpanel.mdx
Comment (joshblack) [inline content/components/selectpanel.mdx:21]: One note for this image, declaritive -> declarative

File: content/components/selectpanel.mdx
Comment (joshblack) [inline content/components/selectpanel.mdx:60]: From a technical perspective, wanted to ask how do we support indeterminate states on an option in listbox?

File: content/components/selectpanel.mdx
Comment (joshblack) [inline content/components/selectpanel.mdx:420]: Random question, for the "instant selection" or simpler cases for selection without a filter, are these situations where we would prefer folks to use components other than a select panel? (like an actionlist or action menu directly)

**Review by joshblack — COMMENTED** [review]
Thanks for taking the time to clean things up and clarify! Just left some comments/questions 👀

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:129]:
```suggestion
1. **Initial** is used during initial data retrieval to avoid displaying an empty list.
2. **Initial Skeleton**is used similarly for initial data retrieval, presenting a skeleton screen instead of an empty list.
3. **Searching/Filtering** uses a minimal loading indicator during searches, allowing users to retain or continue their selections.
4. **Confirmation spinner** is essential when updates aren't instant to indicate the progress of the update.

```

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:140]: This section should move back up to the options category of the document and be replaced with the opening behavior that was originally here. 

```suggestion
<img
  width="960"
  alt="Mouse cursor hovering over an icon button and a focused icon button with the space and enter keys visually below the icon button."
  src="https://user-images.githubusercontent.com/40274682/228910118-3d99fd0a-7b14-4e46-af48-f69bf76912e5.png"
/>

The select panel can be activated through a semantic button, providing mouse users with the ability to click on it, while keyboard users can activate it by pressing the space or enter key.

Additionally, the select panel can be triggered through an action menu. In this scenario, it is important for the action menu to promptly close as soon as the select panel is triggered.

<img
  width="960"
  alt="First select panel has focus applied to the search input with a blue 2px border. Second select panel has no visible focus ring as focus is brought to the H1 of the dialog."
  src="https://github.com/primer/design/assets/980622/fd6a6028-2402-48cb-9d28-693c54f79ac5"
/>

Focus should be placed on the first interactive item within select panel.
```

File: content/components/selectpanel.mdx
Comment (ichelsea) [reply content/components/selectpanel.mdx:144]: @siddharthkp that sounds right, because they're all dialogs.

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:150]: Why remove the helpful videos? Seeing it in action is useful for those who struggle to pull a visual from the words.

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:160]: I don't know why this was removed, it's helpful information especially since we mention that it's dependent on the trigger source. 

```suggestion
The select panel can appear as either an anchored or centered dialog, depending on the trigger source.

- **Anchored dialog**: When triggered by a [button](/components/button) or [icon button](/components/icon-button) but never by an [action menu](/components/action-menu).
- **Centered dialog**: When triggered by a [button](/components/button), [icon button](/components/icon-button) or [action menu](/components/action-menu).

On smaller breakpoints the select panel should always appear as a modal dialog.
```

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:168]: I don't know that this belongs in a note; all of the other notes on the page reference other components.

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:182]:
```suggestion
Ensure clarity in the multi-selection process by using declarative actions. Avoid automatic updates or dialog closure upon selection.
```

File: content/components/selectpanel.mdx
Comment (ichelsea) [reply content/components/selectpanel.mdx:193]: I'm okay with leaving this out. What we're going with is the true and expected nature of a dialog, which is covered in the "Closing" section of this doc. (we should put the videos back in though)

File: content/components/selectpanel.mdx
Comment (ichelsea) [reply content/components/selectpanel.mdx:299]: Yep!

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:335]:
```suggestion
1. In single and multi-selection modes with a search/filter input, an [x-circle-fill](/foundations/icons/x-circle-fill-16) icon button for quick clearing.
2. For multi-selection with search, a 'deselect all' [icon button](/components/icon-button) in the header.
3. In multi-selection without search, a header checkbox allows for one-click selection or deselection of all items.
```

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:356]:
```suggestion
In multi-selections with search functionality, a **deselect all** [icon button](/components/icon-button) becomes available for easy deselection as soon as one item is selected.
```

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:413]: Why are we removing this section? I have a feeling we'll get questions on what is possible if we don't include this.

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:541]: Would you like me to re-record an updated version of this instead of removing it entirely? Selection following focus is often a confused concept.

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:418]: We don't need this since the use case of declarative actions is already well covered in the document. This also mentions single select which confuses our guidance earlier in the document.
```suggestion
```

File: content/components/selectpanel.mdx
Comment (ichelsea) [inline content/components/selectpanel.mdx:420]: I think this is similar to my comment on the previous paragraph in that we don't need it because the declarative actions use cases are well covered prior to this. I also don't understand what this paragraph is trying to say, it's vague without examples and a bit lengthy.

**Review by ichelsea — COMMENTED** [review]
Sorry this is lengthy but it's important 🙂 

#### Search/filter
This was true prior to the edits but we refer to the input indeterminately as filter or search; we should probably stay consistent with one of them, whichever we pick. 

(`search` is generally understood to look through a database that isn't visually presented in front of the user to bring them results that match. `filter` takes a visually existing dataset and narrows down the results for the user. In our case, we have a little bit of both in that we show users a partial data set but you can search for ones that aren't present in the list that we load)

File: content/components/selectpanel.mdx
Comment (dipree) [reply content/components/selectpanel.mdx:58]: > and a select panel can't be built without a trigger.

I'm not sure I understand the deliberation here. What would be the alternative, to render it on page load?

**Pull request closed without merging**
